### ISSUE TITLE
[FEATURE] Add `.eggignore` file and allow users to chose between the ignore file or the `files` field in egg.json

### ISSUE THREAD

**Issue by MartonDev** [nestlandofficial/nest.land#91]
**Is your feature request related to a problem? If so, please describe.**
Right now the nest.land CLI publishes only the files given in the `files` field, in `egg.json`. Sometimes people have a lot of files imported from various directories and it's hard to keep track of all of them.

Also if you create a new file it is extra time to add it to the `files` field.

**Describe the solution you'd like**
The best solution would be to let the users choose between the `files` field or a `.eggignore` file. If the `.eggignore` file extists, the cli uses it instead of the `files` field.

**Describe alternatives you've considered**
An alternative already exists, using `*`

**Comment by justablob** [comment]
We considered this before releasing, but decided against an .eggignore file in favor of the more explicit `files` field. We considered the case that a developer might forget to ignore a new file, and if that file is supposed to not be published, like a `.env` file, it could cause more damage to publish that one file than to see you forgot to add a file, add it real quick and publish again. If you want to, feel free to add a .gitignore-like file, but please make it work alongside the current `files` field, if that's okay for you. If you really wanna go ignore-based, you can just add a `**/*` glob, as you said already.

**Comment by martonlederer** [comment]
Yes I will make sure to make it work alongside the `files` field. As I mentioned, it will be optional, so `eggs` will use the `files` field by default, and will only use the ignore file, if it exists and has content

**Comment by zorbyte** [comment]
I propose we set it as optional. Instead set it up to use the files field and if there is no files field then look for a .eggignore. In addition, the eggignore should be applied on top of the files field if present. In the case neither are present, then do not upload any files.

**Comment by maximousblk** [comment]
how about just use the `.gitignore` file but there will be a commented section under which the files for eggs will be listed?

example:

```gitignore
.idea
.env
*.log

# /*eggignore

# docs
# examples

# */eggignore
```

**Comment by t8** [comment]
@maximousblk Thanks for the idea! Although, I will say that we're working towards staying away from relying on any sort of source control so that we don't limit the scope of the registry.

**Comment by maximousblk** [comment]
Can we use `deps.ts` as an alternative to egg.json?

like just export an object named `egg` which eggs could import and work with.

example:

```js
// ./mod.ts

export const egg = {
  name: "audit",
  version: "1.0.0",
  description: "Add a notification for insecure versions of your modules.",
  stable: true,
  unlisted: false,
  repository: "https://github.com/maximousblk/audit",
  files: ["./mod.ts", "./README.md"],
};
```

**Comment by jsejcksn** [comment]
It would be nice to provide or link to a utility for generating the files array from the package directory. I think the main chore here is having to manually create the list. +1 for the util supporting a filter file like `.gitignore`

**Comment by t8** [comment]
@jsejcksn I completely agree. Adding some "intelligence" to this feature would greatly improve UX. We'll certainly consider implementing this in a future release. Thanks for bringing this up!
